### DISCUSSION TITLE
Branches and Rulesets

### DISCUSSION THREAD

**Discussion by itsnotmik** [DrAlzahraniProjects/csusb_fall2024_cse6550_team1#3] [Ideas]
Hello,

It is probably a good idea to set up some different branches and set up rulesets for pull requests and merging. This will hopefully help in our development process and prevent any unexpected errors from improper github actions.

In terms of branches, there are many different methods for when branches should be made, but I believe that a simple approach of having a DEVELOP and a MAIN branch is probably fine enough for the project. I would recommend looking at different branching structures/strategies to identify the most appropriate for our project.

In terms of rulesets, I believe this will be super helpful in preventing unwanted merging to the wrong branch, and help prevent accidental merging of code that is nonoperational. Look into the branch protection rules and other rulesets in the repository settings under Rules.

Thank you

**Comment by Boganabi** [comment]
I was thinking this too. I created a rule for that :)
